### PR TITLE
Fix up build, get bijection-util onto 2.12

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- Removes the always-enabled-coverage from the build (we don't need `sbt-doge` for 2.11 or 2.10, so we can use the old style in `.travis.yml`
- formats plugins
- upgrades `util-core` for 2.12 only
- adds caching to the build
- adds better badges to README.md
- adds a notification webhook for gitter to the .travis.yml
 - pulls some versions out into `val`s
- removes old doc generation (I'll make a microsite soon)

### CODE REVIEW

**Comment by sritchie** [comment]
Cc @johnynek

File: build.sbt [hunk 1]
```diff
@@ -249,8 +261,8 @@ lazy val bijectionScrooge = {
     libraryDependencies ++= Seq(
       "org.apache.thrift" % "libthrift" % "0.6.1" exclude ("junit", "junit"),
```
Comment (johnynek) [inline build.sbt:262]: I wonder if we could move this to util now. @mosesn? Seems like a module there, and/or finagle is the right place.

Comment (johnynek) [reply build.sbt:262]: I guess this is for Scrooge, but you get the idea.

**Review by johnynek — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (mosesn) [reply build.sbt:262]: DM me and we can chat about it.

**Pull request merged**